packages/next/src/utils.ts

```diff
@@ -2157,23 +2163,44 @@ export {
   getSourceFilePathFromPage,
 };
 
-interface MiddlewareManifest {
+type MiddlewareManifest = MiddlewareManifestV1 | MiddlewareManifestV2;
+
+interface MiddlewareManifestV1 {
   version: 1;
   sortedMiddleware: string[];
-  middleware: { [page: string]: EdgeFunctionInfo };
-  functions?: { [page: string]: EdgeFunctionInfo };
+  middleware: { [page: string]: EdgeFunctionInfoV1 };
+  functions?: { [page: string]: EdgeFunctionInfoV1 };
+}
+
+interface MiddlewareManifestV2 {
+  version: 2;
+  sortedMiddleware: string[];
+  middleware: { [page: string]: EdgeFunctionInfoV2 };
+  functions?: { [page: string]: EdgeFunctionInfoV2 };
 }
```

Comment:
**thought:** what a shame typescript does not allow us extending interfaces and overriding some of the fields

Reply:
yeah, we can do that if we use `Omit<>` to exclude overriding fields but not sure if it's better in this case.